### PR TITLE
[Feature]Feature/arjun/subscription

### PR DESCRIPTION
**Pull request by arj9912**

### CODE REVIEW

File: subscription/models.py [hunk 1]
```diff
@@ -3,6 +3,10 @@
 from cooperative.models import Finance
 
 
+class PlanCost(BaseModelMixin):
+    price = models.DecimalField(max_digits=10, decimal_places=2)
+    recurrance_period = models.DateField()
```
Comment (RahulRimal) [inline subscription/models.py:8]: move this to Subscription

File: subscription/serializers.py [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
 from subscription.models import Plan, PlanCost, Subscription
 
 
-class PlanSerializer(BaseModelSerializerMixin):
+class PlanCostSerializer(BaseModelSerializerMixin):
     class Meta:
-        model = Plan
-        fields = ["idx", "name", "description", "period"]
+        model = PlanCost
+        fields = ["idx", "price", "recurrance_period"]
```
Comment (RahulRimal) [inline subscription/serializers.py:12]: remove recurrance_period

File: subscription/serializers.py [hunk 3]
```diff
@@ -27,6 +40,19 @@ class Meta:
             "next_billing",
             "grace_period",
             "status",
-            "verified",
+            "payment_verified",
             "auto_renewable",
         ]
```
Comment (RahulRimal) [inline subscription/serializers.py:45]: add recurrance_period too

File: subscription/serializers.py [hunk 4]
```diff
@@ -27,6 +40,19 @@ class Meta:
...
+            finance = Finance.objects.get(idx=validated_data.pop("finance_idx"))
+        except Finance.DoesNotExist:
+            raise serializers.ValidationError("Finance does not exist")
+        validated_data["finance"] = finance
+        try:
+            plan = Plan.objects.get(idx=validated_data.pop("plan_idx"))
+        except Plan.DoesNotExist:
+            raise serializers.ValidationError("Plan does not exist")
+        validated_data["plan"] = plan
+        return super().create(validated_data)
```
Comment (RahulRimal) [inline subscription/serializers.py:58]: calculate recurrance_period and then add to subscription

**Review by RahulRimal — CHANGES_REQUESTED** [review]

File: subscription/models.py (hunk 1 above)
Comment (arj9912) [reply subscription/models.py]: ok done

File: subscription/serializers.py (hunk 2 above)
Comment (arj9912) [reply subscription/serializers.py]: ok done

File: subscription/serializers.py (hunk 3 above)
Comment (arj9912) [reply subscription/serializers.py:49]: ok done

File: subscription/serializers.py (hunk 4 above)
Comment (arj9912) [reply subscription/serializers.py:63]: ok done

**Review by RahulRimal — APPROVED** [review]

**Pull request merged**